cmd/kube-apiserver/app/server.go

```diff
@@ -460,15 +460,17 @@ func BuildGenericConfig(s *options.ServerRunOptions, proxyTransport *http.Transp
 	}
 
 	webhookAuthResolver := func(delegate webhookconfig.AuthenticationInfoResolver) webhookconfig.AuthenticationInfoResolver {
-		return webhookconfig.AuthenticationInfoResolverFunc(func(server string) (*rest.Config, error) {
+		return webhookconfig.AuthenticationInfoResolverFunc(func(server string, directRouting bool) (*rest.Config, error) {
 			if server == "kubernetes.default.svc" {
 				return genericConfig.LoopbackClientConfig, nil
 			}
-			ret, err := delegate.ClientConfigFor(server)
+			ret, err := delegate.ClientConfigFor(server, directRouting)
 			if err != nil {
 				return nil, err
 			}
-			if proxyTransport != nil && proxyTransport.Dial != nil {
```

Comment:
We should find a different way to signal this special handling for kube apiserver than polluting the interface for all consumers

Reply:
An alternative is to check whether the `server` is a `[host|ip]:port` or not, but that seems a little bit hacky.